source/common/router/upstream_request.cc

```diff
@@ -128,9 +129,9 @@ UpstreamRequest::UpstreamRequest(RouterFilterInterface& parent,
     parent_.callbacks()->activeSpan().injectContext(trace_context, upstream_context);
   }
 
-  stream_info_.setUpstreamInfo(std::make_shared<StreamInfo::UpstreamInfoImpl>());
+  stream_info_.setUpstreamInfo(upstream_info_);
```

Comment:
why create this early but not set it early?  Worth code comments I think

Reply:
This PR actually just set the upstream host early. The upstream info is still be created at the constructor (we just moved it more the constructor function to the constructor initialize list)?